axelor-account/src/main/java/com/axelor/apps/account/service/fixedasset/FixedAssetValidateServiceImpl.java

```diff
@@ -0,0 +1,111 @@
+package com.axelor.apps.account.service.fixedasset;
+
+import com.axelor.apps.account.db.FixedAsset;
+import com.axelor.apps.account.db.repo.FixedAssetRepository;
+import com.axelor.common.StringUtils;
+import com.axelor.db.JPA;
+import com.axelor.exception.AxelorException;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+
+public class FixedAssetValidateServiceImpl implements FixedAssetValidateService {
+
+  protected FixedAssetLineService fixedAssetLineService;
+
+  protected FixedAssetGenerationService fixedAssetGenerationService;
+
+  protected FixedAssetDerogatoryLineService fixedAssetDerogatoryLineService;
+
+  protected FixedAssetRepository fixedAssetRepo;
+
+  @Inject
+  public FixedAssetValidateServiceImpl(
+      FixedAssetLineService fixedAssetLineService,
+      FixedAssetGenerationService fixedAssetGenerationService,
+      FixedAssetDerogatoryLineService fixedAssetDerogatoryLineService,
+      FixedAssetRepository fixedAssetRepo) {
+    this.fixedAssetLineService = fixedAssetLineService;
+    this.fixedAssetGenerationService = fixedAssetGenerationService;
+    this.fixedAssetDerogatoryLineService = fixedAssetDerogatoryLineService;
+    this.fixedAssetRepo = fixedAssetRepo;
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @throws AxelorException
+   * @throws NullPointerException if fixedAsset is null
+   */
+  @Override
+  @Transactional(rollbackOn = {Exception.class})
+  public void validate(FixedAsset fixedAsset) throws AxelorException {
+    Objects.requireNonNull(fixedAsset);
+    if (fixedAsset.getGrossValue().compareTo(BigDecimal.ZERO) > 0) {
+
+      if (StringUtils.isEmpty(fixedAsset.getFixedAssetSeq())) {
+        fixedAsset.setFixedAssetSeq(fixedAssetGenerationService.generateSequence(fixedAsset));
+      }
+
+      if (fixedAsset.getFixedAssetLineList() != null
+          && !fixedAsset.getFixedAssetLineList().isEmpty()) {
+        fixedAssetLineService.clear(fixedAsset.getFixedAssetLineList());
+      }
+      if (fixedAsset.getFiscalFixedAssetLineList() != null
+          && !fixedAsset.getFiscalFixedAssetLineList().isEmpty()) {
+        fixedAssetLineService.clear(fixedAsset.getFiscalFixedAssetLineList());
+      }
+      if (fixedAsset.getFixedAssetDerogatoryLineList() != null
+          && !fixedAsset.getFixedAssetDerogatoryLineList().isEmpty()) {
+        fixedAssetDerogatoryLineService.clear(fixedAsset.getFixedAssetDerogatoryLineList());
+      }
+      if (fixedAsset.getIfrsFixedAssetLineList() != null
+          && !fixedAsset.getIfrsFixedAssetLineList().isEmpty()) {
+        fixedAssetLineService.clear(fixedAsset.getIfrsFixedAssetLineList());
+      }
+
+      if (!fixedAsset
+          .getDepreciationPlanSelect()
```

Comment:
can this field be null instead of empty ? I'm not sure because it is not required but in AOP a string can be nullable (the generated getter does not return an empty string. So I think it is better to check if the field is null before.

Reply:
Should not be null today, but managed null value as it could be null in the futur